lib/spack/docs/packaging_guide.rst

```diff
@@ -2521,6 +2521,100 @@ would do with the package's own spec:
 
    mpicc = join_path(my_mpi.prefix.bin, 'mpicc')
 
+^^^^^^^^^^^^^^^^^^^^
+Spec build interface
+^^^^^^^^^^^^^^^^^^^^
+
+For more complex queries Spack supports an extended sub-scripting syntax:
+
+.. code-block:: python
+
+   mpi_cxx_and_fortran_query = spec['mpi:cxx,fortran']
+
+which allows for a list of comma separated values to be specified after
+the package name (a single colon separates the package name from the values).
+These values will be attached to the ``Spec`` object being created and can be
+used by the dependency to fine tune the computation of certain attributes:
+
+.. code-block:: python
+
+   # Get the list of MPI Fortran libraries
+   mpi_fortran_libs = spec['mpi:fortran'].libs
```

Comment:
Because you lose a uniform and modular access to the properties. What you should do if a package provides both `mpi` and `lapack`? What if it makes sense to add a new parameter, say `ilp64`? Do we need to provide an attribute for every combination like:
```python
spec['mpi'].cxx_libs
spec['mpi'].fortran_libs
spec['mpi'].ilp64_libs
spec['mpi'].cxx_ilp64_libs
spec['mpi'].fortran_ilp64_libs
spec['mpi'].cxx_fortran_ilp64_libs
```
?